src/Service.Tests/Caching/DabCacheServiceIntegrationTests.cs

```diff
@@ -252,6 +253,90 @@ await Assert.ThrowsExceptionAsync<DataApiBuilderException>(
+
+            // Act
+            int cacheEntryTtl = 1;
+            JsonArray? result = await dabCache.GetOrSetAsync<JsonArray>(executeQueryAsync: mockExecuteQuery.Object, queryMetadata: queryMetadata, cacheEntryTtl: cacheEntryTtl);
+
+            // Assert
+            Assert.AreEqual(expected: true, actual: mockExecuteQuery.Invocations.Count is 1, message: ERROR_UNEXPECTED_INVOCATIONS);
+
+            // Validates that the expected database response is returned by the cache service.
+            Assert.AreEqual(expected: expectedDatabaseResponse, actual: result, message: ERROR_UNEXPECTED_RESULT);
+        }
+
+        /// <summary>
+        /// Tests DAB's cache service invocation when the type is JsonArray.
+        /// JsonArray aligns with the type used for executing stored procedures against
+        /// MSSQL databases.
+        /// This test validates that a cache hit occurs when the same request
+        /// is submitted before the cache entry expires. Validates that
+        /// DabCacheService.CreateCacheKey(..) outputs the same key given constant input.
+        /// </summary>
+        [TestMethod]
+        public async Task JsonArray_CacheServiceInvocation_CacheHit_NoFactoryInvocation()
+        {
+            // Arrange
+            using FusionCache cache = CreateFusionCache(sizeLimit: 1000, defaultEntryTtlSeconds: 1);
+            JsonArray? expectedDatabaseResponse = new()
+            {
+                JsonNode.Parse(@"{""key"": ""value""}"),
+                JsonNode.Parse(@"{""key"": ""value2""}")
+            };
+
+            Mock<Func<Task<JsonArray>>> mockExecuteQuery = new();
+            mockExecuteQuery.Setup(e => e.Invoke()).Returns(Task.FromResult(expectedDatabaseResponse));
+
+            Dictionary<string, DbConnectionParam> parameters = new()
+            {
+                {"param1", new DbConnectionParam(value: "param1Value") }
+            };
+
+            DatabaseQueryMetadata queryMetadata = new(queryText: "select c.name from c", dataSource: "dataSource1", queryParameters: parameters);
+            DabCacheService dabCache = CreateDabCacheService(cache);
+
+            int cacheEntryTtl = 1;
+            // First call. Cache miss
+            _ = await dabCache.GetOrSetAsync<JsonArray>(executeQueryAsync: mockExecuteQuery.Object, queryMetadata: queryMetadata, cacheEntryTtl: cacheEntryTtl);
+
+            // Act
+            JsonArray? result = await dabCache.GetOrSetAsync<JsonArray>(executeQueryAsync: mockExecuteQuery.Object, queryMetadata: queryMetadata, cacheEntryTtl: cacheEntryTtl);
+
+            // Assert
+            Assert.AreEqual(expected: true, actual: mockExecuteQuery.Invocations.Count is 1, message: ERROR_UNEXPECTED_INVOCATIONS);
+            Assert.IsFalse(mockExecuteQuery.Invocations.Count > 1, message: "Expected a cache hit, but observed cache misses.");
+            Assert.AreEqual(expected: true, actual: mockExecuteQuery.Invocations.Count is 1, message: ERROR_UNEXPECTED_INVOCATIONS);
```

Comment:
redundant assert here on line 336?

Reply:
i saw one other instance of this. Removing

Reply:
fixed